python/cuml/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -828,3 +861,39 @@ class RandomForestClassifier(BaseRandomForestModel,
         if self.dtype == np.float64:
             return get_rf_json(rf_forest64).decode('utf-8')
         return get_rf_json(rf_forest).decode('utf-8')
+
+    def cpu_to_gpu(self):
+        # treelite does an internal isinstance check to detect an sklearn
+        # RF, which proxymodule interferes with. We work around that
+        # temporarily here just for treelite internal check and
+        # restore the __class__ at the end of the method.
```

Comment:
In a future version of Treelite, we should add an alternative version of `treelite.sklearn.import_model` that takes in a second argument `model_type`? The caller would provide a string to uniquely identify the model type, so that we can remove the internal `isinstance` check.

Reply:
Maybe, depends on whether we do other changes in cuML itself to potentially avoid this